pkg/commands/terraform.go

```diff
@@ -34,54 +25,49 @@ func addTerraformCmd(topLevel *cobra.Command) {
 		Short:  `Terraform check-divergence check if there are drifts in the state.`,
 		PreRun: upgradeIfNotLatest,
 		Run: func(cmd *cobra.Command, args []string) {
-			terraformCheckDivergence(&options)
-		},
-	}
-
-	addCommonFlags(checkDivergence, &options)
-	rootCmd.AddCommand(checkDivergence)
-	topLevel.AddCommand(rootCmd)
-}
-
-// terraformCheckDivergence tell us if there is a divergence in our terraform plan.
-// This basically translates in: "is there any changes pending to apply?"
-func terraformCheckDivergence(o *terraformOptions) error {
-
-	contextLogger := log.WithFields(log.Fields{"subcommand": "check-divergence"})
+			contextLogger := log.WithFields(log.Fields{"subcommand": "check-divergence"})
 
-	terraform := terraform.Commander{}
+			contextLogger.Info("Executing terraform plan, if there is a drift this program execution will fail")
+			err := options.CheckDivergence()
 
-	if o.displayTfOutput {
-		terraform.DisplayOutput = true
-	}
-
-	var TfVarFile []string
-
-	// Check if user provided a terraform var-file
-	if o.varFile != "" {
-		TfVarFile = append([]string{fmt.Sprintf("-var-file=%s", o.varFile)})
+			if err != nil {
+				contextLogger.Fatal("Error executing plan, either an error or a divergence")
+			}
+		},
 	}
 
-	contextLogger.Info("Executing terraform plan, if there is a drift this program execution will fail")
+	apply := &cobra.Command{
+		Use:    "apply",
```

Comment:
What's your thinking behind having Terraform apply here? I'm using wondering where it would be used..

Reply:
The idea I have on my mind is to add a second step after the [divergence pipeline](https://concourse.cloud-platform.service.justice.gov.uk/teams/main/pipelines/divergence) which can be triggered manually and would run `terraform apply`. Similar to what they document [here](https://concourse-ci.org/manual-trigger-example.html)

Reply:
I also think in the near future, when we start to execute everything in a pipeline, we will need to execute terraform apply in different places. We already have the CLI which does a plan (divergence is just a plan).

Reply:
Sounds good to me.